In [6]:
"""Assignment-02b: Exploratory Data Analysis with Pandas
Context:
You are a data analyst tasked with performing exploratory data analysis (EDA)
on the Titanic dataset. This dataset contains information about passengers
aboard the Titanic, including demographics and survival status.
Your goal is to analyze the dataset using Pandas to gain insights into the
passengers and their characteristics."""
#Task 1: Loading the Data:
# Load the Titanic dataset into a Pandas DataFrame.
import pandas as pd

auto = pd.read_csv(r"C:\Users\Admin\Desktop\Swaroopa\Lithan_Phyton\el02_fc02_Assignment02\titanic.csv")
auto.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [7]:
"""Task 2: Initial Exploration:"""
#Display the first 5 rows of the dataset to get an overview.
auto.head(6)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S
5,6,0,3,"Moran, Mr. James",male,NaN,0,0,330877,8.4583,NaN,Q


In [8]:
# Provide basic information about the dataset using the info() method.
auto.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [9]:
#Task 3: Handling Missing Values:
#Identify and count the number of missing values in each column.
auto.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [26]:
# Decide on an appropriate strategy for handling missing values in at least
#two columns and implement it.

auto['Embarked'] = auto['Embarked'].fillna(auto['Embarked'].mode()[0])
auto['Cabin'] = auto['Cabin'].fillna(auto['Cabin'].mode()[0])

In [23]:
# Verify that missing values have been handled
print("\nMissing values after handling:")
print(auto.isnull().sum())


Missing values after handling:
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            891
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin            0
Embarked         0
dtype: int64


In [27]:
"""Task 4: Descriptive Statistics:"""
#Generate descriptive statistics for the numerical columns using the describe() method.
auto.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,0.0,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,NaN,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,NaN,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,NaN,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,NaN,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,NaN,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,NaN,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,NaN,8.000000,6.000000,512.329200


In [28]:
"""Task 5: Unique Values:"""
#List unique values in the 'Pclass' column.
auto['Pclass'].unique()


array([3, 1, 2])

In [29]:
#Count the occurrences of each unique value in the 'Embarked' column.
auto['Embarked'].value_counts()

Embarked
S    646
C    168
Q     77
Name: count, dtype: int64

In [30]:
"""Task 6: Data Manipulation:"""
#Create a new column named 'FamilySize' that represents the sum of 'SibSp' and 'Parch'.
FamilySize = auto['SibSp'] + auto['Parch']
auto['FamilySize'] = FamilySize
auto.head()



,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked,FamilySize
0,1,0,3,"Braund, Mr. Owen Harris",male,NaN,1,0,A/5 21171,7.2500,B96 B98,S,1
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,NaN,1,0,PC 17599,71.2833,C85,C,1
2,3,1,3,"Heikkinen, Miss. Laina",female,NaN,0,0,STON/O2. 3101282,7.9250,B96 B98,S,0
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,NaN,1,0,113803,53.1000,C123,S,1
4,5,0,3,"Allen, Mr. William Henry",male,NaN,0,0,373450,8.0500,B96 B98,S,0


In [31]:
#Drop the 'Cabin' column from the dataset.
auto.drop('Cabin', axis=1, inplace=True)
auto.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Embarked,FamilySize
0,1,0,3,"Braund, Mr. Owen Harris",male,NaN,1,0,A/5 21171,7.2500,S,1
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,NaN,1,0,PC 17599,71.2833,C,1
2,3,1,3,"Heikkinen, Miss. Laina",female,NaN,0,0,STON/O2. 3101282,7.9250,S,0
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,NaN,1,0,113803,53.1000,S,1
4,5,0,3,"Allen, Mr. William Henry",male,NaN,0,0,373450,8.0500,S,0


In [32]:
"""Task 7: Filtering and Sorting:"""
# Create a new DataFrame containing only passengers with an age greater than 30.
newage30 = auto[auto['Age']>30]
newage30.head(6)


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Embarked,FamilySize


In [67]:
#Sort the DataFrame by 'Fare' in descending order.
auto.sort_values(by='Fare', ascending=False)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Embarked,FamilySize
679,680,1,1,"Cardeza, Mr. Thomas Drake Martinez",male,36.0,0,1,PC 17755,512.3292,C,1
258,259,1,1,"Ward, Miss. Anna",female,35.0,0,0,PC 17755,512.3292,C,0
737,738,1,1,"Lesurer, Mr. Gustave J",male,35.0,0,0,PC 17755,512.3292,C,0
88,89,1,1,"Fortune, Miss. Mabel Helen",female,23.0,3,2,19950,263.0000,S,5
438,439,0,1,"Fortune, Mr. Mark",male,64.0,1,4,19950,263.0000,S,5
...,...,...,...,...,...,...,...,...,...,...,...,...
806,807,0,1,"Andrews, Mr. Thomas Jr",male,39.0,0,0,112050,0.0000,S,0
815,816,0,1,"Fry, Mr. Richard",male,28.0,0,0,112058,0.0000,S,0
466,467,0,2,"Campbell, Mr. William",male,28.0,0,0,239853,0.0000,S,0
481,482,0,2,"Frost, Mr. Anthony Wood ""Archie""",male,28.0,0,0,239854,0.0000,S,0


In [33]:
"""Task 8: Data Analysis:"""
#Calculate the survival rate for different passenger classes ('Pclass').
auto.groupby('Pclass')['Survived'].mean()



Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64

In [34]:
#Identify and display the passenger with the highest 'Fare'
auto[auto['Fare'] == auto['Fare'].max()]

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Embarked,FamilySize
258,259,1,1,"Ward, Miss. Anna",female,NaN,0,0,PC 17755,512.3292,C,0
679,680,1,1,"Cardeza, Mr. Thomas Drake Martinez",male,NaN,0,1,PC 17755,512.3292,C,1
737,738,1,1,"Lesurer, Mr. Gustave J",male,NaN,0,0,PC 17755,512.3292,C,0


"""Conclusion:
•	Write a brief conclusion summarizing key findings and observations from the analysis."""

The Titanic dataset contains information about passengers, with some missing values in the 'Age', 'Cabin', and 'Embarked' columns. We handled the missing values in 'Age' by filling them with the median and in 'Embarked' by filling them with the mode. The 'Cabin' column had a large number of missing values and was dropped.

A new column 'FamilySize' was created by combining 'SibSp' and 'Parch', representing the total number of siblings, spouses, parents, and children aboard for each passenger.

The survival rate varies significantly across passenger classes. Passengers in Class 1 had the highest survival rate (approximately 63%), followed by Class 2 (approximately 47%), and then Class 3 (approximately 24%). This suggests that passenger class played a significant role in survival likelihood.

We also identified the passenger(s) with the highest fare, which was 512.3292. There were multiple passengers with this highest fare.

Overall, the analysis provides initial insights into the characteristics of the Titanic passengers and factors that may have influenced survival.

